packages/react-native/Libraries/promiseRejectionTrackingOptions.js

```diff
@@ -22,7 +24,26 @@ let rejectionTrackingOptions: $NonMaybeType<Parameters<enable>[0]> = {
       // $FlowFixMe[method-unbinding] added when improving typing for this parameters
       message = Error.prototype.toString.call(rejection);
       const error: Error = (rejection: $FlowFixMe);
-      stack = error.stack;
+
+      const warning =
+        `Possible Unhandled Promise Rejection (id: ${id}):\n` +
+        `${message ?? ''}\n` +
+        (stack == null ? '' : stack);
+
+      // Print pretty unhandled rejections while on DEV
+      if (__DEV__) {
+        LogBox.addLog({
+          level: 'warn',
+          message: {content: warning, substitutions: []},
+          componentStack: [],
+          stack: error.stack,
+          category: 'possible_unhandled_promise_rejection',
+        });
+
+        return;
+      } else {
+        console.warn(warning);
+      }
```

Comment:
We never write to `stack` anymore, can you cleanup `warning`?

And I think this will log twice in `!__DEV__` (although this module is only ever used `if (__DEV__)`, so we could remove this check altogether really.